src/acquisition/covid_hosp/common/database.py

```diff
@@ -172,6 +172,33 @@ def insert_dataset(self, publication_date, dataframe):
                        tuple(values) +
                        tuple(i[0] for i in self.additional_fields))
 
+    if hasattr(self, 'AGGREGATE_KEY_COLS'):
+      ak_cols = self.AGGREGATE_KEY_COLS
+
+      # restrict data to just the key columns and remove duplicate rows
+      ak_data = dataframe[ak_cols].drop_duplicates()
+      # cast types (bools and NULLables)
+      for col in ak_cols:
+        if col.startswith('is_'): # TODO: this is hacky af
+          ak_data[col] = (ak_data[col] == 'true')
+      ak_data = ak_data.to_numpy(na_value=None).tolist()
+
+      # create string of tick-quoted and comma-seperated column list
+      ak_cols_str = ','.join([ '`'+col+'`' for col in ak_cols ])
+      # ...and ticked and comma-sep'd "column=column" list for ON UPDATE (to keep only the most recent values for each pk)
+      ak_updates_str = ','.join([ '`'+col+'`=`'+col+'`' for col in ak_cols ])
```

Comment:
tracking lots of nested delimiters is hard to read -- maybe this instead?

```suggestion
      ak_cols_str = ','.join(f'`{col}`' for col in ak_cols)
      # ...and ticked and comma-sep'd "column=column" list for ON UPDATE (to keep only the most recent values for each pk)
      ak_updates_str = ','.join(f'`{col}`=`{col}`' for col in ak_cols)
```